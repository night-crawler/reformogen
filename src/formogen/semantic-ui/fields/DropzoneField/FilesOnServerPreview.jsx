import React from 'react';
import PropTypes from 'prop-types';
import { isArray } from 'lodash';
import { List, Segment } from 'semantic-ui-react';

import { FileOnServerItem } from './FileOnServerItem';


FilesOnServerPreview.propTypes = {
  isRemovable: PropTypes.bool,
  files: PropTypes.any,
  onDeleteFile: PropTypes.func,
};
FilesOnServerPreview.defaultProps = {
  isRemovable: false,
  files: [],
  onDeleteFile: (fileUrl) => // eslint-disable-next-line
    console.warn('FilesOnServerPreview.onDeleteFile', fileUrl),
};
export function FilesOnServerPreview(props) {
  // TODO: it looks like a pipe dream, there's no way for multiple files to be stored in one field
  const files = isArray(props.files) 
    ? props.files 
    : [ props.files ];

  return (
    <Segment as={ List } attached='bottom' divided={ true } relaxed={ true }>
      { files.map((fileUrl, i) => 
        <FileOnServerItem 
          key={ i } 
          fileUrl={ fileUrl } 
          isRemovable={ props.isRemovable }
          onDelete={ props.onDeleteFile }  
        />
      ) }
    </Segment>
  );
}
