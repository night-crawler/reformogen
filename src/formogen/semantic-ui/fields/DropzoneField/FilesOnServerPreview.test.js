import { FilesOnServerPreview } from './FilesOnServerPreview';

it('<FilesOnServerPreview />', () => {
  const wrapper = shallow(
    <FilesOnServerPreview
      files={ [ 'https://domain.tld/1.png' ] }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
