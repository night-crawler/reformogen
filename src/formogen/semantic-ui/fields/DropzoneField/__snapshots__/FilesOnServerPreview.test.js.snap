// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilesOnServerPreview /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilesOnServerPreview
    files={
      Array [
        "https://domain.tld/1.png",
      ]
    }
    isRemovable={false}
    onDeleteFile={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": [Function],
      "attached": "bottom",
      "children": Array [
        <FileOnServerItem
          fileUrl="https://domain.tld/1.png"
          getPreviewImage={[Function]}
          isRemovable={false}
          onDelete={[Function]}
        />,
      ],
      "divided": true,
      "relaxed": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "fileUrl": "https://domain.tld/1.png",
          "getPreviewImage": [Function],
          "isRemovable": false,
          "onDelete": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "attached": "bottom",
        "children": Array [
          <FileOnServerItem
            fileUrl="https://domain.tld/1.png"
            getPreviewImage={[Function]}
            isRemovable={false}
            onDelete={[Function]}
          />,
        ],
        "divided": true,
        "relaxed": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "fileUrl": "https://domain.tld/1.png",
            "getPreviewImage": [Function],
            "isRemovable": false,
            "onDelete": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
